pkg/cmd/server/origin/run_components.go

```diff
@@ -516,3 +519,20 @@ func (c *MasterConfig) RunClusterQuotaReconciliationController() {
 	c.ClusterQuotaMappingController.GetClusterQuotaMapper().AddListener(controller)
 	go controller.Run(5, utilwait.NeverStop)
 }
+
+// RunIngressIPController starts the ingress ip controller if IngressIPNetworkCIDR is configured.
+func (c *MasterConfig) RunIngressIPController(client *kclient.Client) {
+	// TODO need to disallow if a cloud provider is configured
+	if len(c.Options.NetworkConfig.IngressIPNetworkCIDR) == 0 {
+		glog.V(2).Infof("Ingress ip controller will not start - IngressIPNetworkCIDR not configured")
```

Comment:
All old ones we want to remove.

On Wed, Aug 3, 2016 at 1:40 PM, Maru Newby <notifications@github.com> wrote:

> In pkg/cmd/server/origin/run_components.go
> <https://github.com/openshift/origin/pull/9454#discussion_r73377275>:
>
> > @@ -516,3 +519,20 @@ func (c *MasterConfig) RunClusterQuotaReconciliationController() {
> >  	c.ClusterQuotaMappingController.GetClusterQuotaMapper().AddListener(controller)
> >  	go controller.Run(5, utilwait.NeverStop)
> >  }
> > +
> > +// RunIngressIPController starts the ingress ip controller if IngressIPNetworkCIDR is configured.
> > +func (c *MasterConfig) RunIngressIPController(client *kclient.Client) {
> > +	// TODO need to disallow if a cloud provider is configured
> > +	if len(c.Options.NetworkConfig.IngressIPNetworkCIDR) == 0 {
> > +		glog.V(2).Infof("Ingress ip controller will not start - IngressIPNetworkCIDR not configured")
>
> Done. You're clearly the authority here, but I did add this message to be
> consistent with some of the other controllers not starting due to missing
> configuration:
>
>
> https://github.com/marun/origin/blob/32c0ad3977d08318cdd13da01152308411e62910/pkg/cmd/server/origin/run_components.go#L90
>
> https://github.com/marun/origin/blob/32c0ad3977d08318cdd13da01152308411e62910/pkg/cmd/server/origin/run_components.go#L109
>
> https://github.com/marun/origin/blob/32c0ad3977d08318cdd13da01152308411e62910/pkg/cmd/server/origin/run_components.go#L417
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9454/files/32c0ad3977d08318cdd13da01152308411e62910#r73377275>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p7IZonM8LM47suosKQd6SBfw7vm_ks5qcNKggaJpZM4I6YVl>
> .
>